scalyr_agent/builtin_monitors/windows_event_log_monitor.py

```diff
@@ -773,6 +801,91 @@ def log_event(self, event):
         finally:
             self._bookmark_lock.release()
 
+    def _param_placeholder_value(self, channel, provider, param):
+        if not re.match("^%%[0-9]+$", param):
+            return param
+
+        if channel == "Security":
+            provider = "Security"
+
+        param_key = "%s-%s-%s" % (channel, provider, param)
+        if param_key in self._param_cache:
+            return self._param_cache[param_key]
+
+        dll_key = "%s-%s" % (channel, provider)
+        if dll_key in self._dll_cache:
+            if not isinstance(self._dll_cache[dll_key], _DLL):
```

Comment:
Ah, if it's not a _DLL instance, then this has to be an exception. Got it. I just was wondering why we put exceptions to the cache :) 
And it seems that there's no way for a cached Exception to make it to the `FormatMessageW `

Reply:
Yes, I should've either commented this or made a note in the PR description.  The idea is to store an exception to avoid attempting to (unsuccessfully) open a dll over and over again.

Reply:
Do we want to have some kind of TTL on the cache items here? In case some library which provides some DLL gets installed later one and future calls would not result in an exception anymore?

Or it's just really an overkill and we can just assume user will also restart the agent in such case or similar?